From: Suresh Siddha <suresh.b.siddha@intel.com>
Date: Tue, 31 Jul 2012 10:29:14 -0700
Subject: [PATCH] x86, avx: don't use avx instructions with "noxsave" boot param
Patch-mainline: v3.6
References: bnc#821802

Clear AVX, AVX2 features along with clearing XSAVE feature bits,
as part of the parsing "noxsave" parameter.

Fixes the kernel boot panic with "noxsave" boot parameter.

We could have checked cpu_has_osxsave along with cpu_has_avx etc, but Peter
mentioned clearing the feature bits will be better for uses like
static_cpu_has() etc.

Signed-off-by: Suresh Siddha <suresh.b.siddha@intel.com>
Link: http://lkml.kernel.org/r/1343755754.2041.2.camel@sbsiddha-desk.sc.intel.com
Cc: <stable@vger.kernel.org>	# v3.5
Signed-off-by: H. Peter Anvin <hpa@zytor.com>
Acked-by: NeilBrown <neilb@suse.de>

Automatically created from "patches.suse/xor-0009-x86-avx-don-t-use-avx-instructions-with-noxsave-boot.patch" by xen-port-patches.py

--- sle11sp3.orig/arch/x86/kernel/cpu/common-xen.c	2013-06-12 12:35:56.000000000 +0200
+++ sle11sp3/arch/x86/kernel/cpu/common-xen.c	2013-06-12 12:38:39.000000000 +0200
@@ -159,6 +159,8 @@ static int __init x86_xsave_setup(char *
 {
 	setup_clear_cpu_cap(X86_FEATURE_XSAVE);
 	setup_clear_cpu_cap(X86_FEATURE_XSAVEOPT);
+	setup_clear_cpu_cap(X86_FEATURE_AVX);
+	setup_clear_cpu_cap(X86_FEATURE_AVX2);
 	return 1;
 }
 __setup("noxsave", x86_xsave_setup);
